Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -740,21 +785,23 @@ def get_characters_features(text):
 
 
 def get_url_features(email_body, email_html, soup):
-    http_regex = r'http://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
-    https_regex = r'https://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
-    http_urls = re.findall(http_regex, email_body)
-    https_urls = re.findall(https_regex, email_body)
-    embedded_urls = [a['href'] for a in soup.findAll('a')]
-    all_urls = http_urls + https_urls + embedded_urls
+    url_regex = r'(?:(?:https?|ftp|hxxps?):\/\/|www\[?\.\]?|ftp\[?\.\]?)(?:[-\w\d]+\[?\.\]?)+[-\w\d]+(?::\d+)?' \
+                r'(?:(?:\/|\?)[-\w\d+&@#\/%=~_$?!\-:,.\(\);]*[\w\d+&@#\/%=~_$\(\);])?'
+    embedded_urls = []
+    for a in soup.findAll('a'):
+        if a.has_attr('href'):
+            embedded_urls.append(a['href'])
+    plain_urls = re.findall(url_regex, email_body)
+    all_urls = plain_urls + embedded_urls
```

Comment:
I think sometimes you can count url twice if it's HTML and also the text is the URL itself, maybe you should dedup this list using set